File: ios/Talise/Features/Rewards/GoalSuccessView.swift
import SwiftUI

/// Full-screen success confirmation shown after adding to a savings goal.
/// Mirrors `SavingsSuccessView` (the invest success) so the two flows feel
/// identical: shared green glow, a hero that drops in with the scrapbook
/// wobble, a display headline, one quiet mono sub-line, and the white
/// "Back to Invest" pill that returns the user to the Invest screen.
///
/// The hero is a target/crosshair — the goal motif — in the brand green.
/// `amountText` is pre-formatted in the user's display currency by the caller.
struct GoalSuccessView: View {
    /// Pre-formatted, currency-aware amount added, e.g. "₦1,000.00".
    let amountText: String
    /// The goal's name, for the sub-line.
    let goalName: String
    let onDismiss: () -> Void

    var body: some View {
        ZStack {
            SuccessGlowBackground()

            VStack(spacing: 0) {
                Spacer()

                // Target hero — the goal motif, in brand green on a soft disc,
                // dropping in with the same scrapbook wobble as the piggy.
                ZStack {
                    Circle()
                        .fill(TaliseColor.accent.opacity(0.14))
                        .frame(width: 184, height: 184)
                    Image(systemName: "target")
                        .font(.system(size: 104, weight: .regular))
                        .foregroundStyle(TaliseColor.accent)
                }
                .scrapbookEntry(delay: 0.05, tilt: -6)

                Spacer().frame(height: 30)

                Text("Getting closer to your target")
                    .font(TaliseFont.display(38, weight: .regular))
                    .kerning(-0.8)
                    .foregroundStyle(TaliseColor.fg)
                    .multilineTextAlignment(.center)
                    .minimumScaleFactor(0.6)
                    .lineLimit(2)
                    .padding(.horizontal, 24)
                    .scrapbookFadeUp(delay: 0.22)

                Text("\(amountText) added to \(goalName).")
                    .font(TaliseFont.mono(13, weight: .regular))
                    .kerning(-0.26)
                    .foregroundStyle(TaliseColor.fgMuted)
                    .multilineTextAlignment(.center)
                    .lineSpacing(2)
                    .fixedSize(horizontal: false, vertical: true)
                    .frame(width: 310)
                    .padding(.top, 14)
                    .scrapbookFadeUp(delay: 0.30)

                Spacer()

                Button(action: onDismiss) {
                    Text("Back to Invest")
                        .font(TaliseFont.body(15, weight: .medium))
                        .kerning(-0.3)
                        .foregroundStyle(.black)
                        .frame(width: 175, height: 41)
                        .background(Capsule().fill(.white))
                }
                .buttonStyle(.plain)
                .padding(.bottom, 40)
                .scrapbookFadeUp(delay: 0.38)
            }
        }
        .preferredColorScheme(.dark)
    }
}

File: android/app/src/main/java/io/talise/app/feature/rewards/GoalSuccessView.kt
package io.talise.app.feature.rewards

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import io.talise.app.R
import io.talise.app.ui.theme.TaliseColors
import io.talise.app.ui.theme.TaliseType

/** Which flow the success screen confirms — iOS `GoalSuccessView.Kind`. */
enum class GoalSuccessKind { Deposit, Withdraw }

/**
 * Full-screen success confirmation shown after adding to (or withdrawing
 * from) a savings goal — iOS `GoalSuccessView`. Shared green-on-black look:
 * the target hero drops in with the scrapbook wobble, a display headline,
 * one quiet mono sub-line, and the white "Back to Invest" pill.
 *
 * `amountText` is pre-formatted by the caller (e.g. "$25.00").
 */
@Composable
fun GoalSuccessView(
    amountText: String,
    goalName: String,
    onDismiss: () -> Unit,
    kind: GoalSuccessKind = GoalSuccessKind.Deposit,
) {
    val headline = if (kind == GoalSuccessKind.Deposit) "Getting closer to your target" else "Back in your balance"
    val subline = if (kind == GoalSuccessKind.Deposit) {
        "$amountText added to $goalName."
    } else {
        "$amountText withdrawn from $goalName."
    }

    Column(
        Modifier
            .fillMaxSize()
            .background(TaliseColors.bg),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Spacer(Modifier.weight(1f))

        // Target hero — the hand-drawn crosshair illustration, dropping in
        // with the same scrapbook wobble as the savings piggy.
        Image(
            painter = painterResource(R.drawable.goaltarget),
            contentDescription = null,
            contentScale = ContentScale.Fit,
            modifier = Modifier
                .size(260.dp)
                .scrapbookEntry(delayMillis = 50, tilt = -6f),
        )

        Spacer(Modifier.height(30.dp))

        Text(
            headline,
            style = TaliseType.display(38.sp, FontWeight.Normal),
            letterSpacing = (-0.8).sp,
            color = TaliseColors.fg,
            textAlign = TextAlign.Center,
            maxLines = 2,
            modifier = Modifier
                .padding(horizontal = 24.dp)
                .scrapbookFadeUp(delayMillis = 220),
        )

        Text(
            subline,
            style = TaliseType.mono(13.sp),
            letterSpacing = (-0.26).sp,
            color = TaliseColors.fgMuted,
            textAlign = TextAlign.Center,
            lineHeight = 17.sp,
            modifier = Modifier
                .width(310.dp)
                .padding(top = 14.dp)
                .scrapbookFadeUp(delayMillis = 300),
        )

        Spacer(Modifier.weight(1f))

        Box(
            Modifier
                .width(175.dp)
                .height(41.dp)
                .background(Color.White, CircleShape)
                .clickable { onDismiss() }
                .scrapbookFadeUp(delayMillis = 380),
            contentAlignment = Alignment.Center,
        ) {
            Text(
                "Back to Invest",
                style = TaliseType.body(15.sp, FontWeight.Medium),
                letterSpacing = (-0.3).sp,
                color = Color.Black,
            )
        }
        Spacer(Modifier.height(40.dp))
    }
}
